src/test/java/unit/BqDatasetLifetime.java

```diff
@@ -0,0 +1,227 @@
+package unit;
+
+import static harness.utils.ExternalBQDatasets.randomDatasetId;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import bio.terra.cli.serialization.userfacing.resource.UFBqDataset;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.cloud.bigquery.Dataset;
+import com.google.cloud.bigquery.DatasetId;
+import harness.TestCommand;
+import harness.baseclasses.SingleWorkspaceUnit;
+import harness.utils.ExternalBQDatasets;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInfo;
+
+/** Tests for specifying default lifetimes for controlled BQ Datasets. */
+@Tag("unit")
+public class BqDatasetLifetime extends SingleWorkspaceUnit {
+
+  @Override
+  @BeforeEach
+  protected void setupEachTime() throws IOException {
+    super.setupEachTime();
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+  }
+
+  @AfterEach
+  protected void cleanupEachTime(TestInfo testInfo) {
+
+    // Assumption is that each test case will create a single dataset, named after the test method
+    // itself.  This will clean up after the test case.
```

Comment:
consider using a `private String datasetName;`, set it in the test, check if it's non-null here and delete it, instead of assuming a name and using TestInfo.

OTOH, why bother? Can the workspace be deleted regardless of if there are datasets in it?

Reply:
Agreed, I think can skip deleting resources after each test as long as they're all named something different. They'll all get cleaned up when the workspace is deleted.

Reply:
Done